apps/backend/src/modules/plans/plan.controller.ts

```diff
@@ -0,0 +1,233 @@
+ *           example: "high"
+ *         userId:
+ *           type: number
+ *           example: 1
+ */
+class PlanController extends BaseController {
+	private planService: PlanService;
+
+	public constructor(logger: Logger, planService: PlanService) {
+		super(logger, APIPath.PLANS);
+
+		this.planService = planService;
+
+		this.addRoute({
+			handler: (options) =>
+				this.findById(options as APIHandlerOptions<{ params: { id: number } }>),
+			method: "GET",
+			path: PlansApiPath.PLAN,
+		});
+
+		this.addRoute({
+			handler: (options) =>
+				this.create(
+					options as APIHandlerOptions<{
+						body: PlanCreateRequestDto;
+					}>,
+				),
+			method: "POST",
+			path: PlansApiPath.PLAN_CREATE,
+			validation: {
+				body: planCreateValidationSchema,
+			},
+		});
+	}
+
+	/**
+	 * @swagger
+	 * /plans/create:
+	 *   post:
+	 *     tags:
+	 *       - plans
+	 *     summary: Create a new plan
+	 *     security:
+	 *       - bearerAuth: []
+	 *     requestBody:
+	 *       required: true
+	 *       content:
+	 *         application/json:
+	 *           schema:
+	 *             $ref: '#/components/schemas/PlanRequestDto'
+	 *     responses:
+	 *       200:
+	 *         description: Plan created successfully
+	 *         content:
+	 *           application/json:
+	 *             schema:
+	 *               $ref: '#/components/schemas/PlanResponseDto'
+	 *       401:
+	 *         description: Unauthorized - Invalid or missing authentication token
+	 *         content:
+	 *           application/json:
+	 *             schema:
+	 *               type: object
+	 *               properties:
+	 *                 message:
+	 *                   type: string
+	 *                   example: "Unauthorized"
+	 */
+	private async create(
+		options: APIHandlerOptions<{ body: PlanCreateRequestDto }>,
+	): Promise<APIHandlerResponse> {
+		return {
+			payload: await this.planService.create(options.body),
+			status: HTTPCode.OK,
+		};
+	}
+
+	/**
+	 * @swagger
+	 * /plans/{id}:
+	 *   get:
+	 *     tags:
+	 *       - plans
+	 *     summary: Get plan by ID
+	 *     security:
+	 *       - bearerAuth: []
+	 *     parameters:
+	 *       - in: path
+	 *         name: id
+	 *         schema:
+	 *           type: number
+	 *         required: true
+	 *         description: ID of the plan
+	 *     responses:
+	 *       200:
+	 *         description: Plan retrieved successfully
+	 *         content:
+	 *           application/json:
+	 *             schema:
+	 *               $ref: '#/components/schemas/PlanDaysTaskDto'
+	 *       401:
+	 *         description: Unauthorized - Invalid or missing authentication token
+	 *         content:
+	 *           application/json:
+	 *             schema:
+	 *               type: object
+	 *               properties:
+	 *                 message:
+	 *                   type: string
+	 *                   example: "Unauthorized"
+	 */
+	private async findById(
+		options: APIHandlerOptions<{ params: { id: number } }>,
```

Comment:
{ id: number } type is used in many places. I realise that it is a simple one but still let's move it to a separate type and reuse